tasks.py

```diff
@@ -985,3 +987,203 @@ def validate_app_config(context):
+        m = re.match(r"<=\s*([0-9.]+)", part)
+        if m:
+            return None, m[1]
+        m = re.match(r"<\s*([0-9.]+)", part)
+        if m:
+            v = m[1]
+            v_parts = [int(x) for x in v.split(".")]
+            if v_parts[1] > 0:
+                max_v = max_version(f"{v_parts[0]}.{v_parts[1]-1}", "minor")
+            else:
+                max_v = max_version(f"{v_parts[0]-1}", "major")
+            return None, max_v
+        return None, None
+
+    constraint = constraint.strip()
+    # Multiple constraints, e.g. ">=2.0.3,<3.0.0"
+    if "," in constraint:
+        min_v = None
+        max_v = None
+        for part in constraint.split(","):
+            part_min, part_max = parse_single_bound(part)
+            if part_min is not None:
+                min_v = part_min
+            if part_max is not None:
+                max_v = part_max
+        return min_v, max_v
+    # Caret ^
+    if constraint.startswith("^"):
+        v = constraint[1:]
+        parts = v.split(".")
+        if int(parts[0]) > 0:
+            max_v = max_version(parts[0], "major")
+        elif int(parts[0]) == 0 and len(parts) > 1 and int(parts[1]) > 0:
+            max_v = max_version(f"0.{parts[1]}", "minor")
+        else:
+            max_v = max_version(v, "patch")
+        return v, max_v
+    # Compatible ~=
+    if constraint.startswith("~="):
+        v = constraint[2:]
+        parts = v.split(".")
+        if len(parts) == 3:
+            min_v = v
+            max_v = max_version(f"{parts[0]}.{parts[1]}", "minor")
+        elif len(parts) == 2:
+            min_v = f"{parts[0]}.{parts[1]}.0"
+            max_v = max_version(f"{parts[0]}.{parts[1]}", "minor")
+        elif len(parts) == 1:
+            min_v = f"{parts[0]}.0.0"
+            max_v = max_version(parts[0], "major")
+        return min_v, max_v
+    # Tilde ~
+    if constraint.startswith("~"):
+        v = constraint[1:]
+        parts = v.split(".")
+        if len(parts) > 1:
+            max_v = max_version(f"{parts[0]}.{parts[1]}", "minor")
+            min_v = v
+            if len(parts) == 2:
+                min_v = f"{parts[0]}.{parts[1]}.0"
+        else:
+            min_v = f"{parts[0]}.0.0"
+            max_v = max_version(parts[0], "major")
+        return min_v, max_v
+    # Wildcard
+    if "*" in constraint:
+        parts = constraint.replace("*", "0").split(".")
+        if len(parts) == 3:
+            min_v = f"{parts[0]}.{parts[1]}.0"
+            max_v = max_version(f"{parts[0]}.{parts[1]}", "minor")
+        elif len(parts) == 2:
+            min_v = f"{parts[0]}.0.0"
+            max_v = max_version(parts[0], "major")
+        # This shouldn't happen, but handle it gracefully
+        elif len(parts) == 1:
+            min_v = "0.0.0"
+            max_v = "*"
+        return min_v, max_v
+    # Exact version or ==<version>
+    # e.g. 2.0.3 or ==2.0.3
+    m = re.match(r"==?\s*([0-9.]+)", constraint)
```

Comment:
```suggestion
    m = re.match(r"(?:==)?\s*([0-9.]+)", constraint)
```

The `?` is only making the previous `=` optional but I think you want both of them to be optional?

Reply:
Yep, good catch. I was trying to not create a grouping, but missed making them both optional.